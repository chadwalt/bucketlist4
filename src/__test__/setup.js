import { configure } from 'enzyme'
import Adapter from "enzyme-adapter-react-15"

configure({ adapter: new Adapter() })

global.window = {}
import localStorage from "mock-local-storage"
window.localStorage = global.localStorage