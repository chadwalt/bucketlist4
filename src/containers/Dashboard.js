/*
* This page will hold the bucketlists.
* *******************************************/

import React, { Component } from "react"
import Buckets from "../components/buckets"
import { Redirect } from "react-router"
import AddBucketModal from "../components/add_bucket_model"
import BaseUrl from "../config"
import axios from "axios"

class Dashboard extends Component {
	constructor(props) {
		super(props)

		this.state = { // Set the current state of this prop.
			bucket_name: "",
			redirect: "",
			model_title: "Add Bucket!"
		}

		this.logout = this.logout.bind(this)
		this.saveBucket = this.saveBucket.bind(this)
		this.handleInput = this.handleInput.bind(this)

	}

	/* This will handle getting input from the form.
	* @param event => Event fired when the input is changed in the form.
	********************************************************************/
	handleInput(event) {
		const name = event.target.name
		const value = event.target.value
		this.setState({ [name]: value })
	}

	logout(e) {
		e.preventDefault()

		// Delete the token in the session Storage.
		sessionStorage.removeItem("auth_token")

		this.setState({ redirect: "logout" })
	}

	/* Save the bucket 
	* *****************/
	saveBucket() {

		if (!this.state.bucket_name) {
			alert("Please provide a name for the bucket.")
			return
		}
		let auth_token = sessionStorage.auth_token // Get the auth_token from the session storage.

		var params = new URLSearchParams() // Using the x-www-form-urlencoded.
		params.append("name", this.state.bucket_name)

		axios({
			method: "POST",
			url: BaseUrl + "bucketlists/",
			data: params,
			headers: { "Authorization": auth_token }
		}).then(function (response) {
			var data = response.data
			if (data.success) {
				alert("Bucketlist Saved successfully")
			}

		})


	}

	render() {
		const { redirect } = this.state

		if (redirect === "logout") {
			return (<Redirect to="/" />) // Redirect to login.
		} else if (redirect === "" || sessionStorage.token) {
			return (
				<div>
					<Buckets
						{...this.state}
						onLogout={this.logout}
						onAddBucket={this.addBucket}
					/>
					<AddBucketModal title={this.state.model_title} bucketName={this.state.bucket_name}
						onSaveBucket={this.saveBucket}
						onInput={this.handleInput}
					/>
				</div>

			)
		}


	}
}



export default Dashboard
