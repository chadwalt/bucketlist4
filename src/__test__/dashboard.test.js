import React from "react"
import Buckets from "../components/buckets"
import renderer from "react-test-renderer"
import axios from "axios"
import MockAdapter from "axios-mock-adapter"
import { StaticRouter } from "react-router"
import Dashboard from "../containers/Dashboard"
import "./setup"
import BaseUrl from "../config"

import { shallow, mount } from "enzyme"
const mock = new MockAdapter(axios)

mock.onGet(BaseUrl + "bucketlists/?page=1&rows=8").reply(200, {})
mock.onPost(BaseUrl + "auth/logout").reply(200, { success: true })
mock.onDelete(BaseUrl + "bucketlists/1").reply(200, { success: true })
mock.onGet(BaseUrl + "bucketlists/?q=Climbing&page=1&rows=8").reply(200, { success: true })
mock.onGet(BaseUrl + "bucketlists/?page=1&rows=8").reply(200, {
	buckets: {
		id: 3,
		name: "Visiting USA."
	}
})

describe("Testing the Dashboard container", () => {

	const mockFunction = jest.fn()
	//let wrapper

	const props = {
		displayTable: "buckets",
		onLogout: mockFunction,
		modelId: "#myModal",
		onEdit: mockFunction,
		onAdd: mockFunction,
		onAddBucket: mockFunction,
		onOpenDelete: mockFunction,
		onSearch: mockFunction,
		pagination: mockFunction,
		onMovePage: mockFunction,
		dismissAlert: mockFunction
	}

	// afterEach(() => {
	// 	wrapper = shallow(<Dashboard {...props} />)
	// 	wrapper.setState({ bucket_name: "Joining the navy" })
	// 	sessionStorage.setItem("auth_token", "382283jjdjasdfadfa")
	// 	mock.onPost(BaseUrl + "bucketlists", wrapper.state().bucket_name).reply(200, { success: true })
	// })

	it("It calls the onclick event for the edit of the bucket.", () => {
		const dashboard = mount(
			<StaticRouter context={{}}>
				<Dashboard {...props} />
			</StaticRouter>
		)
		expect(dashboard).toHaveLength(1)
	})

	it("renders correctly", () => {
		const rendered = renderer.create(
			<StaticRouter context={{}}>
				<Dashboard {...props} />
			</StaticRouter>)
		expect(rendered.toJSON()).toMatchSnapshot()
	})

	it("It should check change state", () => {
		const wrapper = shallow(<Dashboard {...props} />)

		sessionStorage.setItem("auth_token", "382283jjdjasdfadfa")
		wrapper.instance().checkLoggedStatus()
		expect(wrapper.state().isLoggedIn).toEqual(true)

		wrapper.instance().onDismiss()
		expect(wrapper.state().visible).toEqual(false)

		wrapper.instance().editBucket("Edit Bucket", 1)
		expect(wrapper.state().model_title).toEqual("Edit Bucket")

		wrapper.instance().addBucket()
		expect(wrapper.state().model_title).toEqual("Add Bucket")

		wrapper.instance().openDeleteBucket("Climbing", 1)
		expect(wrapper.state().bucket_name).toEqual("Climbing")

		wrapper.instance().componentDidMount()
	})

	it("it should logout user", () => {
		const wrapper = shallow(<Dashboard {...props} />)
		const event = {
			preventDefault: () => { }
		}

		wrapper.instance().logout(event)
		expect(wrapper.state().redirect).toEqual("logout")
	})

	it("It should update state on form input.", () => {
		const wrapper = shallow(<Dashboard {...props} />)
		const event = {
			preventDefault: () => { },
			target: {
				name: "buckets",
				value: "Climbing"
			}
		}

		wrapper.instance().handleInput(event)
		expect(wrapper.state().buckets).toEqual("Climbing")
	})

	it("It should delete bucket.", () => {	
		const wrapper = shallow(<Dashboard />)
		sessionStorage.setItem("auth_token", "382283jjdjasdfadfa")	

		wrapper.setState({bucket_id: 1})
		wrapper.instance().deleteBucket()
		expect(wrapper.state().buckets.length).toEqual(0)
	})

	it("It should move to next page.", () => {	
		const wrapper = shallow(<Dashboard {...props} />)
		sessionStorage.setItem("auth_token", "382283jjdjasdfadfa")	

		wrapper.instance().movePage(1)
		expect(wrapper.state().buckets.length).toEqual(0)
	})

	it("It should search through the buckets.", () => {	
		const wrapper = shallow(<Dashboard {...props} />)
		sessionStorage.setItem("auth_token", "382283jjdjasdfadfa")	

		const event = {
			preventDefault: () => { },
			target: {
				value: "Climbing"
			}
		}

		wrapper.instance().searchBucket(event)
		expect(wrapper.state().buckets.length).toEqual(0)
	})

})
