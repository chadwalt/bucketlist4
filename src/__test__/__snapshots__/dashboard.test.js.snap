// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Dashboard container renders correctly 1`] = `null`;
